### PR TITLE
3356 Fix sms TypeError: Cannot read property 'profile' of undefined

### LINKED ISSUE #3356: Error occurred while sending sms TypeError: Cannot read property 'profile' of undefined
Orders can complete, but are delayed while this error occurs.

```
23:46:32.639Z  WARN Reaction:
  Error: error occured while sending sms TypeError: Cannot read property 'profile' of undefined
      at DDPCommon.MethodInvocation.smsSend (imports/plugins/included/sms/server/methods/sms.js:52:30)
      at packages/check.js:128:16
      at Meteor.EnvironmentVariable.EVp.withValue (packages/meteor.js:1184:15)
      at Object._failIfArgumentsAreNotAllChecked (packages/check.js:127:41)
      at maybeAuditArgumentChecks (packages/ddp-server/livedata_server.js:1765:18)
      at DDP._CurrentMethodInvocation.withValue (packages/ddp-server/livedata_server.js:1686:15)
      at Meteor.EnvironmentVariable.EVp.withValue (packages/meteor.js:1184:15)
      at resolve (packages/ddp-server/livedata_server.js:1684:36)
      at new Promise (<anonymous>)
      at Server.applyAsync (packages/ddp-server/livedata_server.js:1683:12)
      at Server.apply (packages/ddp-server/livedata_server.js:1622:26)
      at Server.call (packages/ddp-server/livedata_server.js:1604:17)
      at DDPCommon.MethodInvocation.notificationSend (imports/plugins/included/notifications/server/methods/notifications.js:57:16)
      at packages/check.js:128:16
      at Meteor.EnvironmentVariable.EVp.withValue (packages/meteor.js:1184:15)
      at Object._failIfArgumentsAreNotAllChecked (packages/check.js:127:41)
      at maybeAuditArgumentChecks (packages/ddp-server/livedata_server.js:1765:18)
      at DDP._CurrentMethodInvocation.withValue (packages/ddp-server/livedata_server.js:719:19)
      at Meteor.EnvironmentVariable.EVp.withValue (packages/meteor.js:1184:15)
      at DDPServer._CurrentWriteFence.withValue (packages/ddp-server/livedata_server.js:717:46)
      at Meteor.EnvironmentVariable.EVp.withValue (packages/meteor.js:1184:15)
      at Promise (packages/ddp-server/livedata_server.js:715:46)
```
### Steps to reproduce the behavior
Placed order with Shippo + PayPal Express + Stripe.  

Configured with rc-config, nothing done specifically to setup SMS.

Place an order as a user.
Login as admin to fullfill order, witness server console.

### Versions
```
Node: 6.11.1
NPM: 5.5.1
Meteor Node: 8.8.1
Meteor NPM: 5.4.2
Reaction CLI: 0.23.2
Reaction: 1.6.0
Reaction branch: meteor-1.6
Docker: 17.09.0-ce
```

### PR DESCRIPTION
**Pull request by iakhator**
resolve #3356

This PR fix a TypeError of profile of undefined. This actually happens whenever send sms is triggered which doesn't find a user in the database thus throwing the profile of undefined error.

#### Fix
Checked that the user exist before calling the profile from the user object else return false.

#### Test
Configure email, shipping, Stripe payment

- Place an order as a user
- go to any mongo client(Robo Mongo) 
- In the Account collection delete the user
- Login as an Admin
- Go to Orders
-  Fulfil order
- Click on Approve then Capture
- Scroll down Click of `all Items packed`
- Observe the green toastr appear sending an email
- Observe server console to see no error.

### CODE REVIEW

**Comment by pmn4** [comment]
lgtm

File: imports/plugins/included/sms/server/methods/sms.js [hunk 1]
```diff
@@ -49,7 +49,7 @@ Meteor.methods({
     check(shopId, String);
 
     const user = Accounts.findOne({ _id: userId });
-    const addressBook = user.profile.addressBook;
+    const addressBook = user ? user.profile.addressBook : false;
```
Comment (zenweasel) [inline imports/plugins/included/sms/server/methods/sms.js]: Shouldn't we check for both `user` and `user.profile` before returning `addressBook`?

Comment (iakhator) [reply imports/plugins/included/sms/server/methods/sms.js]: I didn't really see the `profile.addressBook` nesting. I was in a rush to fix the profile of undefined error not knowing there will be another undefined of addressBook if profile is not available. I will do this check ASAP @zenweasel

**Comment by zenweasel** [comment]
Tested. Verified fixed.

**Pull request merged**